### ISSUE TITLE
"Normal" Training Loss and Strange Test Result

### ISSUE THREAD

**Issue by Owen-Liuyuxuan** [nianticlabs/manydepth#4]
After fixing the "--png" bug, I also faced difficulties in reproducing good results.

## Training Loss
![manydepth_loss](https://user-images.githubusercontent.com/31618608/117086547-644aaf00-ad7f-11eb-85a4-b81659f6c165.png)

with command 
```
CUDA_VISIBLE_DEVICES=0 python3 -m manydepth.train --data_path /home/kitti_raw/ --log_dir workdirs/ --model_name manydepth --png
```

which is quite normal (I don't know what it is expected to be but that is reasonable at least).

## Test Results
```
   abs_rel |   sq_rel |     rmse | rmse_log |       a1 |       a2 |       a3 |                                                                                                                              
&   0.454  &   4.961  &  12.336  &   0.607  &   0.288  &   0.541  &   0.754  \\ 
```

which is of course a wrong one.

## Tensorboard Validation events
![tensorboard_val_color](https://user-images.githubusercontent.com/31618608/117086945-a6c0bb80-ad80-11eb-8551-84470bd81e12.png)
![tensorboard_val_disp](https://user-images.githubusercontent.com/31618608/117086947-a7f1e880-ad80-11eb-8fce-20e3ef975bbd.png)

I can't detect bug from here.

## Local Modification of Codes
For codes, I modified the datasets/mono_dataset.py on the color augmentation part in compatibility with the new torchvision (which does not seems to be the main problem).
![git_diff_dataset](https://user-images.githubusercontent.com/31618608/117087583-7f6aee00-ad82-11eb-8c75-c72b3501768f.png)

I also modified the export_gt scripts (I don't find the original script works because the splits are on the upper level folder of the script).

![git_diff_export_gt](https://user-images.githubusercontent.com/31618608/117087454-0ff4fe80-ad82-11eb-95cb-4fabc1fec213.png)

**Comment by JamieWatson683** [comment]
Hi - thanks again for your interest!

Yes those scores do not look right. Can you please post a screenshot of the "disp_multi" images too? That is what will be evaluated at test time. Similarly, can you screenshot the "lowest_cost" images?

If those both look sensible, then perhaps it's a bug at evaluation time - can you run your eval command with `--save_pred_disps`, and visualise a few of the disparity estimates? It could be that again blank images are being loaded at test time. 

Thanks!

**Comment by Owen-Liuyuxuan** [comment]
> Hi - thanks again for your interest!
> 
> Yes those scores do not look right. Can you please post a screenshot of the "disp_multi" images too? That is what will be evaluated at test time. Similarly, can you screenshot the "lowest_cost" images?
> 
> If those both look sensible, then perhaps it's a bug at evaluation time - can you run your eval command with `--save_pred_disps`, and visualise a few of the disparity estimates? It could be that again blank images are being loaded at test time.
> 
> Thanks!

## Problem solved
Many thanks for your fast response. Your second point is absolutely correct. I forcefully change the img_ext default arguments to ".png".

I finally got 

```
abs_rel |   sq_rel |     rmse | rmse_log |       a1 |       a2 |       a3 |
&   0.100  &   0.789  &   4.547  &   0.180  &   0.896  &   0.963  &   0.982  \\
```
which is reasonably better than monodepth2 and close to the paper's result. Many thanks for the great work!!! :)

## Minor Suggestions
Though it followed monodepth2, I strongly suggested adding a warning in the readme about this "png/jpg" problem, because PNG is still the default.

Also allowing blank images in training/testing is still rather strange, at least a warning is needed?

**Comment by JamieWatson683** [comment]
Hi,

Sorry for the delay in getting back to you - yes I agree completely, I will push a fix for this shortly.

Thanks again!

**Comment by JamieWatson683** [comment]
Pushed a fix now - will close this issue. Thanks for your help with this!
